HVSample/src/main/java/com/microsoft/hsg/android/hvsample/WeightActivity.java

```diff
@@ -30,112 +37,123 @@
 import android.widget.Toast;
 import android.widget.AdapterView.OnItemSelectedListener;
 
+import healthvault.client.implementation.MicrosoftHealthVaultRestApiImpl;
+import healthvault.client.models.ActionPlan;
+import healthvault.client.models.ActionPlansResponseActionPlanInstance;
+
+import healthvault.client.models.Objective;
+import okhttp3.ResponseBody;
+import retrofit2.Response;
+import rx.Subscriber;
+import rx.Subscription;
+import rx.schedulers.Schedulers;
+
 public class WeightActivity extends Activity {
 
 	private HealthVaultApp service;
-    private HealthVaultClient hvClient;
-    private Record currentRecord;
-    
-    @Override
-    public void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        setContentView(R.layout.weight);
-        service = HealthVaultApp.getInstance();
-        hvClient = new HealthVaultClient();
-        
-        Button weightsBtn = (Button) findViewById(R.id.addWeight);
-        final EditText editText = (EditText) findViewById(R.id.weightInput);
+	private HealthVaultClient hvClient;
+	private Record currentRecord;
+
+	@Override
+	public void onCreate(Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		setContentView(R.layout.weight);
+		service = HealthVaultApp.getInstance();
+		hvClient = new HealthVaultClient();
+
+		final Button weightsBtn = (Button) findViewById(R.id.addWeight);
+		final EditText editText = (EditText) findViewById(R.id.weightInput);
 		
-        weightsBtn.setOnClickListener(new View.OnClickListener() {
-            public void onClick(View view) {
-            	if (service.isAppConnected()) {
-            		putWeight(editText.getText().toString());
-            	}
-            }
-        });
-    }
-    
+		weightsBtn.setOnClickListener(new View.OnClickListener() {
+			public void onClick(View view) {
+				if (service.isAppConnected()) {
+					putWeight(editText.getText().toString());
+				}
+			}
+		});
+	}
+
 	@Override
 	protected void onStart() {
 		super.onStart();
 		hvClient.start();
 	}
-    
-    @Override
-    protected void onResume()
-    {
-        super.onResume();
-        currentRecord = HealthVaultApp.getInstance().getCurrentRecord();
-        getWeights();
-    }
-    
-    @Override
+
+	@Override
+	protected void onResume()
+	{
+		super.onResume();
+		currentRecord = HealthVaultApp.getInstance().getCurrentRecord();
```

Comment:
Do we always need to populate currentRecord before calling getWeights()? If so should we either put the currentRecord assignment in getWeights, or at least have a method which always does currentRecord = ... and getWeights();? Would be better to enforce this strictly rather than have it be something people just need to remember to do.